### PR TITLE
8297130: ComboBox popup doesn't close after selecting value that was added with 'runLater'

### PR DESCRIPTION
**Pull request by mstr2**
This PR fixes a bug where multi-level focus is not correctly preserved.
The original implementation incorrectly assumed that there can only be a single focused node in the scene graph, which is not the case when a branch of the scene graph is presented by a `PopupWindow`. More specifically, when a focused node was removed from the scene graph, the focus flags of all parents were incorrectly cleared. The correct implementation only clears the `focusWithin` flag of parents (but not `focused` or `focusVisible`), and stops when another focused node is encountered along the way.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8297130](https://bugs.openjdk.org/browse/JDK-8297130): ComboBox popup doesn't close after selecting value that was added with 'runLater'

### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - Author)
 * [Ajit Ghaisas](https://openjdk.org/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx pull/956/head:pull/956` \
`$ git checkout pull/956`

Update a local copy of the PR: \
`$ git checkout pull/956` \
`$ git pull https://git.openjdk.org/jfx pull/956/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 956`

View PR using the GUI difftool: \
`$ git pr show -t 956`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/956.diff">https://git.openjdk.org/jfx/pull/956.diff</a>

</details>

### CODE REVIEW

**Comment by kevinrushforth** [comment]
Reviewers: @aghaisas @andy-goryachev-oracle

**Review by andy-goryachev-oracle — APPROVED** [review]
no ill effects observed with several test applications.

incidentally, this change has no effect on JDK-8292933 dual focus issue.

File: modules/javafx.graphics/src/main/java/javafx/scene/Node.java [hunk 1]
```diff
@@ -8184,15 +8184,16 @@ final void notifyFocusListeners() {
...
+     * focus, we need to stop clearing the focusWithin bits if we encounter a parent node that is focused.
      */
-    private void updateRemovedParentFocus(Node oldParent) {
+    private void clearParentsFocusWithin(Node oldParent) {
         if (oldParent != null && focusWithin.get()) {
             Node node = oldParent;
-            while (node != null) {
-                node.focused.set(false);
-                node.focusVisible.set(false);
+            while (node != null && !node.focused.get()) {
```
[the remark this answers is not in the document]
Comment (mstr2) [reply modules/javafx.graphics/src/main/java/javafx/scene/Node.java:8196]: In this area of the code, the eagerly initialized fields `focused`, `focusVisible` and `focusWithin` are used directly. Other than that, there's no reason. If you prefer the convenience getters, I can change it.

**Review by aghaisas — APPROVED** [review]
The fix looks good to me!

**Comment by przemSzer** [comment]
Do you plan to add this fix to the version 19, of Java FX?

**Pull request closed without merging**